packages/esm-ward-app/src/ward-view/ward-view.component.tsx

```diff
@@ -45,20 +47,55 @@ const WardView = () => {
 };
 
 const WardViewByLocation = ({ location }: { location: Location }) => {
-  const { admissionLocation, isLoading, error } = useAdmissionLocation(location.uuid);
+  const {
+    admissionLocation,
+    isLoading: isLoadingLocation,
+    error: errorLoadingLocation,
+  } = useAdmissionLocation(location.uuid);
+  const {
+    admittedPatients,
+    isLoading: isLoadingPatients,
+    error: errorLoadingPatients,
+  } = useAdmittedPatients(location.uuid);
   const { t } = useTranslation();
+  const admittedPatientsByUuid = useMemo(() => {
+    const map = new Map<string, AdmittedPatient>();
+    for (const admittedPatient of admittedPatients ?? []) {
+      map.set(admittedPatient.patient.uuid, admittedPatient);
+    }
+    return map;
+  }, [admittedPatients]);
 
-  if (admissionLocation) {
+  if (admissionLocation != null && admittedPatients != null) {
     const bedLayouts = filterBeds(admissionLocation);
 
     return (
       <>
         {bedLayouts.map((bedLayout, i) => {
           const { patients } = bedLayout;
           const bed = bedLayoutToBed(bedLayout);
+          const patientInfos: WardPatient[] = patients.map((patient) => {
+            const admittedPatient = admittedPatientsByUuid.get(patient.uuid);
 
-          // TODO: replace visit field with real value fetched from useAdmissionLocation (or replacement API)
-          const patientInfos = patients.map((patient) => ({ patient, visit: null }));
+            if (admittedPatient) {
+              // ideally, we can just use the patient object within admittedPatient
+              // and not need the one from bedLayouts, however, the emr api
```

Comment:
This has been fixed and committed in bedmanagement 6.0.0-SNAPSHOT, so you should be able to use admittedPatient now with an appropriate REST call.